### PR TITLE
[REVIEW] Fix to_host issue with column_view having offset

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
to_host was using `head` rather than `data` while accessing column data.

### CODE REVIEW

**Comment by rgsl888prabhu** [comment]
@trevorsm7 @karthikeyann

File: cpp/tests/utilities/column_utilities.hpp [hunk 1]
```diff
@@ -99,7 +99,7 @@ std::vector<bitmask_type> bitmask_to_host(cudf::column_view const& c);
 template <typename T>
 std::pair<std::vector<T>, std::vector<bitmask_type>> to_host(column_view c) {
   std::vector<T> host_data(c.size());
-  CUDA_TRY(cudaMemcpy(host_data.data(), c.head<T>(), c.size() * sizeof(T), cudaMemcpyDeviceToHost));
+  CUDA_TRY(cudaMemcpy(host_data.data(), c.data<T>(), c.size() * sizeof(T), cudaMemcpyDeviceToHost));
   return { host_data, bitmask_to_host(c) };
```
Comment (karthikeyann) [inline cpp/tests/utilities/column_utilities.hpp]: `bitmask_to_host` still copies from element `0`.  
It is possible to copy using `copy_bitmask`. (but device-device copy, then device-host copy involved)

File: cpp/tests/utilities/column_utilities.cu [hunk 2]
```diff
@@ -141,9 +141,10 @@ void expect_equal_buffers(void const* lhs, void const* rhs,
 std::vector<bitmask_type> bitmask_to_host(cudf::column_view const& c) {
   if (c.nullable()) {
     auto num_bitmasks = bitmask_allocation_size_bytes(c.size()) / sizeof(bitmask_type);
+    auto mask = copy_bitmask(c.null_mask(), c.offset(), c.offset()+c.size());
```
Comment (karthikeyann) [inline cpp/tests/utilities/column_utilities.cu]: Just one optimization: 
if `c.offset()` is zero, you can directly use cudaMemCpy with `c.null_mask()`
```cpp
auto mask = (c.offset()==0 ? c.null_mask() : 
copy_bitmask(c.null_mask(), c.offset(), c.offset()+c.size()));
```

Comment (rgsl888prabhu) [reply cpp/tests/utilities/column_utilities.cu]: The type differs, copy_bitmask returns `rmm::device_buffer` and null_mask returns `cudf::bitmask_type *`. If I just want to access data from `rmm::device_buffer`, I have to create an object from copy_bitmask which defeats the change we are trying to make.

Comment (karthikeyann) [reply cpp/tests/utilities/column_utilities.cu]: Type mismatch can be addressed. 
```cpp
rmm::device_buffer mask;
auto dev_bitmask; //use this for cudaMemcpy
if (c.offset()==0) {
   dev_bitmask=c.null_mask();
} else {
      mask = copy_bitmask(c.null_mask(), c.offset(), c.offset()+c.size()));
      dev_bitmask = mask.data(); 
} 
```

**Comment by rgsl888prabhu** [comment]
rerun tests

**Comment by harrism** [comment]
Fixes #3599

**Pull request merged**